### PR TITLE
PLT-2936 Fixed double joined channel messages upon /join

### PR DESCRIPTION
**Pull request by DavidLu1997**
Previous: Joining a channel with /join sends two "X has joined the channel" messages
Expected: Joining a channel with /join sends one "X has joined the channel" message

Unit tests removed because of changed functionality.

### CODE REVIEW

**Comment by jwilander** [comment]
Looks like the same SpinnerButton test issue HH, was having. Can you try upping the timeout like he did [here](https://github.com/mattermost/platform/pull/3013/files#diff-1bacddab05aa04bafb27ab90131d571dR9)?

File: api/command_join_test.go [hunk 1]
```diff
@@ -41,18 +41,7 @@ func TestJoinCommands(t *testing.T) {
...
-
-	found := false
-	for _, c := range c1.Channels {
-		if c.Name == channel2.Name {
-			found = true
-			break
-		}
-	}
-	if !found {
+	if len(c1.Channels) != 5 { // 4 because of town-square, off-topic and direct
```
Comment (crspeller) [inline api/command_join_test.go]: Can we fix this comment?

**Comment by crspeller** [comment]
Can we get an explanation of the fix? From the diff it doesn't make much sense.

File: api/command_join.go [hunk 2]
```diff
@@ -42,18 +42,11 @@ func (me *JoinProvider) DoCommand(c *Context, channelId string, message string)
 
 			if v.Name == message {
 
-				if v.Type == model.CHANNEL_DIRECT {
+				if v.Type != model.CHANNEL_OPEN {
 					return &model.CommandResponse{Text: c.T("api.command_join.fail.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+				} else {
```
Comment (jwilander) [inline api/command_join.go]: No need for the `else` here since it's the end of the function as you're returning. Can you remove it?

**Comment by DavidLu1997** [comment]
The fix is done by removing the actual `JoinChannel` function from `command_join.go`. Previously, that functions sends a Joined Channel message. The command then navigates to the channel link, sending another message. As only open channels were meant to be joined with this command, it simply suffices to navigate to the channel's URL if it exists.

File: api/command_join_test.go [hunk 3]
```diff
@@ -41,18 +41,7 @@ func TestJoinCommands(t *testing.T) {
...
-
-	found := false
-	for _, c := range c1.Channels {
-		if c.Name == channel2.Name {
-			found = true
-			break
-		}
-	}
-	if !found {
+	if len(c1.Channels) != 5 { // 5 because of town-square, off-topic, 2 direct channels, and 1 created channel
```
Comment (crspeller) [inline api/command_join_test.go]: I don't see two direct channels. I count town-square, off-topic, channel0, channel2 and the direct channel.

**Comment by jwilander** [comment]
Will work with DL to resubmit

**Pull request closed without merging**